website/models.py

```diff
@@ -61,3 +61,31 @@ def createUserProfile(sender, instance, created, **kwargs):
 @receiver(post_save, sender=User)
 def saveUserProfile(sender, instance, **kwargs):
     instance.profile.save()
+
+
+@receiver(post_delete, sender=Profile)
+def deleteOnDelete(sender, instance, **kwargs):
+    if instance.photo:
+        if os.path.basename(instance.photo.name) != "defaultuser.png":
+            if os.path.isfile(instance.photo.path):
+                os.remove(instance.photo.path)
+
+@receiver(pre_save, sender=Profile)
+def deleteOnChange(sender, instance, **kwargs):
+    if not instance.pk:
+        return False
+
+    try:
+        oldPhoto = Profile.objects.get(pk=instance.pk).photo
+    except Profile.DoesNotExist:
+        return False
+
+    newPhoto = instance.photo
+    if os.path.basename(oldPhoto.name) == "defaultuser.png":
+        return False
+    else:
+        if os.path.basename(oldPhoto.name) == os.path.basename(newPhoto.name):
+            return False
+
+        if os.path.isfile(oldPhoto.path):
```

Comment:
Would it make sense to refactor the code and pull the common code out into a helper function? The common code being testing if the file is 'defaultuser.png' and testing if it exists, and removing it. And then the new try/catch around that.